### PR TITLE
Clean up Swift main thread assertion

### PR DESCRIPTION
**Pull request by michaelkirk-signal**
PTAL @charlesmchen 

For 2.30

### CODE REVIEW

File: SignalServiceKit/src/Util/OWSSwiftUtils.swift [hunk 1]
```diff
@@ -20,3 +20,16 @@ public func owsFail(_ message: String) {
     Logger.flush()
     assertionFailure(message)
 }
+
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+@objc
+public func AssertIsOnMainThread(file: String = #file,
```
Comment (michaelkirk-signal) [inline SignalServiceKit/src/Util/OWSSwiftUtils.swift]: substantive change 2: add this swift version. (all other changes are either a straight search/replace auto-formatting)

We can leverage the `#file`/`#function`/`#line` as default attributes to make a cleaner call site.

(i.e. `AssertIsOnMainThread()` vs `AssertIsOnMainThread(#function)`)

I'm considering doing something similar for `owsFail` and `Logger.foo` methods so we can stop doing the `"\(logTag) in \(#function)"` dance, but that would require a slightly more hands on approach. WDYT @charlesmchen?

File: SignalServiceKit/src/Util/OWSAsserts.h [hunk 2]
```diff
@@ -151,9 +151,6 @@ __attribute__((annotate("returns_localized_nsstring"))) static inline NSString *
     return s;
 }
 
-// This function is intended for use in Swift.
-void SwiftAssertIsOnMainThread(NSString *functionName);
```
Comment (michaelkirk-signal) [inline SignalServiceKit/src/Util/OWSAsserts.h]: substantive change 1: remove this function

File: SignalServiceKit/src/Util/OWSSwiftUtils.swift (hunk 1 above)
Comment (charlesmchen-signal) [reply SignalServiceKit/src/Util/OWSSwiftUtils.swift]: Noice

> I'm considering doing something similar for owsFail and Logger.foo methods so we can stop doing the "\(logTag) in \(#function)" dance, but that would require a slightly more hands on approach. WDYT @charlesmchen?

I support this.  

I assume `#file/#function/#line` work in free functions and don't capture a reference to self if used in a block.

Comment (charlesmchen-signal) [reply SignalServiceKit/src/Util/OWSSwiftUtils.swift]: Can this method be called from obj-c code?  If so, I suggest we call it `SwiftAssertIsOnMainThread` to help visually distinguish it from `OWSAssertIsOnMainThread()`.

**Comment by charlesmchen-signal** [comment]
> I'm considering doing something similar for owsFail and Logger.foo methods so we can stop doing the "\(logTag) in \(#function)" dance, but that would require a slightly more hands on approach. WDYT @charlesmchen?

I'm going to work on the first part of this (`owsFail()`) since I'm already touching that function.

File: SignalServiceKit/src/Util/OWSSwiftUtils.swift [hunk 3]
```diff
@@ -20,3 +20,16 @@ public func owsFail(_ message: String) {
     Logger.flush()
     assertionFailure(message)
 }
+
+// Once we're on Swift4.2 we can mark this as inlineable
+// @inlinable
+@objc
```
Comment (charlesmchen-signal) [inline SignalServiceKit/src/Util/OWSSwiftUtils.swift]: @michaelkirk I'm getting a build error doing something similar in the "prod fail" methods.

> @objc can only be used with members of classes, @objc protocols, and concrete extensions of classes

File: SignalServiceKit/src/Util/OWSSwiftUtils.swift (hunk 1 above)
Comment (michaelkirk-signal) [reply SignalServiceKit/src/Util/OWSSwiftUtils.swift]: I removed the `@objc`. That was a mistake, I only intended it for Swift.

**Pull request merged**